### ISSUE TITLE
uriparser: update to 0.9.0 (security)

### ISSUE THREAD

**Issue by MingcongBai** [AOSC-Dev/aosc-os-abbs#1432]
**Descriptions:**

```
  * Fixed: Out-of-bounds write in uriComposeQuery* and uriComposeQueryEx*
      Commit 864f5d4c127def386dd5cc926ad96934b297f04e
      Thanks to Google Autofuzz team for the report!
  * Fixed: Detect integer overflow in uriComposeQuery* and uriComposeQueryEx*
      Commit f76275d4a91b28d687250525d3a0c5509bbd666f
      Thanks to Google Autofuzz team for the report!
  * Fixed: Protect uriResetUri* against acting on NULL input
      Commit f58c25069cf4a986fe17a80c5b38687e31feb539
```

**Patches:** N/A

**Architectural progress:**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] ARMv7 `armel`
- [x] PowerPC 64-bit BE `ppc64`
- [x] PowerPC 32-bit BE `powerpc`

**Comment by MingcongBai** [comment]
Ref: #1428

**Comment by MingcongBai** [comment]
Fixed with commit referenced at #1428. Closing.

**Comment by l2dy** [comment]
Use AOSA-2018-0457.
